---
bibliography: manuscript.bib
csl: nature.csl
output:
  github_document: default
  pdf_document:
    dev: png
---

```{r setup, cache=TRUE, message=F, echo=FALSE, warning=F}

	knitr::opts_chunk$set( echo = TRUE )

```

# Multiple pairwise comparisons between species strongly mislead functional genomics studies

Casey W. Dunn^1^*, Felipe Zapata^1,2^, Catriona Munro^1^, Stefan Siebert^1,3^, Andreas Hejnol^4^

^1^ Department of Ecology and Evolutionary Biology, Brown University, Providence, RI, USA

^2^ Current address: Department of Ecology and Evolutionary Biology, University of California Los Angeles, Los Angeles, CA, USA

^3^ Current address: Department of Molecular & Cellular Biology, University of California at Davis, Davis, CA, USA

^4^ Sars International Centre for Marine Molecular Biology, University of Bergen, Bergen, Norway


\* Corresponding author, casey_dunn@brown.edu


## Abstract

Functional genomics, the study of how genomes work, is one of the most rapidly advancing fields in biology. There is now considerable interest in comparing genome function across species to understand what is conserved, what is variable, and how the diversity of organism traits (including anatomy, physiology, and development) relates to genome diversity. Most published comparisons of genome function across species have relied on multiple pairwise comparisons between species, an approach that neglects to account for the evolutionary relationships among species. This has two potential negative impacts. First, it can lead to the wrong conclusions about the evolution of genome function. Second, it is a missed opportunity to learn about biology that can only be understood in an explicit historical context that considers these relationships. Here we examine two recently published comparative gene expression studies that used multiple pairwise comparison rather than phylogenetic approaches to investigate evolutionary patterns of gene expression. We find problems with the pairwise comparisons used in both studies that undermine their published conclusions. The first study we examine found that expression tends to be more different between paralogs than orthologs, and interpreted this as evidence of the ortholog conjecture- the hypothesis that there are greater rates of evolutionary change following duplication. Our phylogenetic analysis finds no difference in rates following duplication and speciation. Instead, the greater difference between paralogs is because they tend to be more distantly related to each other than orthologs are to each other. The second study claimed to find more evolutionary variance in gene expression at a midpoint of embryonic development than earlier and later when comparing distantly related animals. We instead find that changes along the branch of a single taxon, the ctenophore, were over counted because they were included in each pairwise comparison with this species. These reanalyses are concrete demonstrations of the inadequacy of pairwise comparisons in comparative genomics studies, including gene expression analyses, and indicates that it will be critical to adopt phylogenetic comparative methods in future work.

## Introduction

The focus of genomics research has quickly shifted from describing genome sequences to functional genomics, the study of how genomes "work" using tools that measure functional attributes such as expression, chromatin state, and transcription initiation. Functional genomics, in turn, is now becoming more comparative- there is great interest in understanding how functional genomic variation across species gives rise to a diversity of development, morphology, physiology, and other phenotypes [@Wray:2013gh]. Comparative Functional Genomics (CFG) analyses are also critical to transferring functional insight across species, and will grow in importance in coming years.

A rich theoretical and statistical methodology of phylogenetic comparative methods have been developed over the last three decades to address the challenges and opportunities of comparisons across species [@Felsenstein:1985ua; @Grafen:1989um; @Chamberlain:2012vx; @Garamszegi:2014ew; @Pagel:1999gc]. These challenges include non-independence of observations across species -- more closely related species will tend to be more similar to each other than distantly related species. Phylogenetic comparative methods have largely been applied to the evolution of morphological and ecological traits, but are just as relevant to CFG. Most CFG studies have abstained from phylogenetic approaches and instead heavily rely on multiple pairwise comparisons (Fig. 1A) between contemporary traits that do not account for the fact that evolutionary relationships explain much of the structure of variation across species. This leaves CFG studies susceptible to serious statistical errors and is a missed opportunity to ask questions that are only accessible in an explicit phylogenetic context.

One reason that CFG has not yet embraced phylogenetic approaches is that it has not yet been concretely demonstrated that pairwise and phylogenetic approaches can lead to different results in CFG studies. In this manuscript we examine this issue. Although the value of phylogenetic approaches has been repeatedly shown in comparative analyses of other types of character data [@Chamberlain:2012vx; @Ricklefs:1996wb], this value is apparently not well known to the functional genomics community since the field has until recently largely focused on within-species analyses. Many initial CFG studies have also been focused on only two species at a time, where the differences between phylogenetic and pairwise approaches are not as pronounced. As the number of species considered in each CFG study grows, the shortcomings of pairwise approaches will become far more acute. 

Most CFG studies to date compare gene expression across species, often between distinct tissues or developmental time points. Gene expression has been a popular target of CFG studies because, with the advent of RNA-Seq and related transcriptome sequencing methods, it is the most readily collected functional genomics trait across diverse species. Here we reconsider two recent comparative analyses of gene expression data [@KryuchkovaMostacci:2016iw; @Levin:2016bp]. 

The first study, Kryuchkova-Mostacci and Robinson-Rechavi (KMRR) 2016 [@KryuchkovaMostacci:2016iw], analyzed multiple vertebrate expression datasets to test the ortholog conjecture - the hypothesis that orthologs tend of have more conserved attributes (specificity of expression across organs in this case) than do paralogs [@Nehrt:2011bm]. Using pairwise comparisons (Fig. 1A), they found lower expression correlation between paralogs than between orthologs and interpreted this as strong support for the ortholog conjecture. 

The second CFG study we evaluate here is Levin *et al.* [@Levin:2016bp]. This study analyzed gene expression through the course of embryonic development for ten animal species, each from a different phylum. They concluded that these animals exhibit an "inverse hourglass" model for the evolution of gene expression, where there is more evolutionary variance in gene expression at a mid phase of development than there is at early and late phases (Fig. 2). This contrasts with the "hourglass" model previously proposed for closely related species [@Kalinka:2010dw]. Levin *et al.* conclude that this provides biological justification for the concept of phyla and may provide a definition of phyla. We previously described concerns with their interpretations of these results [@Hejnol:2016hw]. Here we directly address the analyses themselves by examining the structure of the pairwise comparisons. 

![Figure Overview](Figure_overview.png?raw=true)

> Figure 1 | Pairwise and phylogenetic comparative approaches. (a) Many comparative functional genomic studies rely on pairwise comparisons, where traits of each gene are compared to traits of other genes (such as orthologs and paralogs). This leads to many more comparisons than unique observations, making each comparison non-independent. (b) Comparative phylogenetic methods, including phylogenetic independent contrasts [@Felsenstein:1985ua], make a smaller number of independent comparisons, where each contrast measures independent changes along different branches. Phylogenetic approaches are rarely used for functional genomics studies. 


## Results and Discussion

### KMRR Reanalysis

#### Original pairwise test of the ortholog conjecture

The KMRR study [@KryuchkovaMostacci:2016iw] sought to test the ortholog conjecture. The ortholog conjecture [@Nehrt:2011bm] is the proposition that orthologs (genes that diverged from each other due to a speciation event) have more similar attributes than do paralogs (genes that diverged from each other due to a gene duplication event). The ortholog conjecture has important biological and technical implications. It shapes our understanding of the functional diversity of gene families. It is used to relate findings from well-studied genes to related genes that have not been investigated in detail. It can be applied to any trait of genes, from gene sequence to biochemical properties to expression. While the ortholog conjecture describes a specific pattern of functional diversity across genes, it is also articulated as a hypothesis about the process of evolution-- that there is greater evolutionary change in gene attributes following a duplication event than a speciation event. 

Despite its importance, there have been relatively few tests of the ortholog conjecture. Previous work has shown that ontology annotations are not sufficient to test the ortholog conjecture [@Chen:2012jz; @Thomas:2012hz]. Analyses of domain structure were consistent with the ortholog conjecture [@Forslund:2011cw]. There have been few tests of the ortholog conjecture with regards to gene expression [@Chen:2012jz], and KMRR is the most thorough such expression study to date.

The KMRR study considered several publicly available datasets. Here we consider their conclusions in the context of one of these studies, Brawand *et al.* 2011 [@Brawand:2011du]. This dataset of gene expression for six organs across ten species (nine mammals and one bird), eight of which were analyzed by KMRR, is the best sampled in their analyses. Their expression summary statistic is Tau [@Yanai:2005dh, @KryuchkovaMostacci:2016in], an indicator of tissue specificity of gene expression. It can range from a value of 0, which indicates no specificity (*i.e.*, uniform expression across tissues), to a value of 1, which indicates high specificity (*i.e.*, expression in only one tissue). It is convenient in that it is a single number of defined range for each gene, though of course since the original expression is multidimensional this means much information is discarded. This includes information about which tissue expression is specific to. For example, if one gene has expression specific to the brain and another expression specific to the kidney, both would have a Tau of 1. 

KMRR's published analyses are based on pairwise comparisons (Figure 1a) between Tau within each gene family. They found the correlation of Tau for orthologs to be significantly greater than the correlation of Tau for paralogs, *i.e.* that orthologs tend to be more similar to each other than paralogs. From this they concluded that their analyses support the ortholog conjecture. They also concluded that this pattern provides support for a particular evolutionary process, that "tissue-specificity evolves very slowly in the absence of duplication, while immediately after duplication the new gene copy differs".

#### Phylogenetic reanalyses

We reanalyzed the KMRR study using phylogenetic comparative methods, focusing on the dataset of Brawand *et al.* 2011 [@Brawand:2011du]. For each internal node in each gene tree (Figure 1b), we measured the phylogenetic independent contrast[@Felsenstein:1985ua] (PIC) of the expression statistic Tau. This is the difference in values of Tau for descendant nodes scaled by the expected variance (determined largely by the lengths of these branches). These contrasts were then annotated by whether they are made across a speciation or duplication event. The original description of independent contrasts [@Felsenstein:1985ua] focused on assessing covariance between changes in two traits on a tree. Our use of contrasts is a bit different-- we look for differences in evolutionary changes of one trait (differential expression) between two categories of nodes (speciation and duplication) on a tree.

![Figure Expectations](figure_expectations.png?raw=true)

> Figure XXExpectations | (a) Under the null hypothesis that there is no difference in the rate of evolution after duplication or speciation events, the distribution of PICs would be the same across both types of nodes. (b) Under the ortholog conjecture, contrasts across duplication nodes would tend to be larger than contrasts across speciation nodes.

To test the ortholog conjecture it is essential to have a null hypothesis that makes predictions distinct from the ortholog conjecture. A suitable null hypothesis is that there is no difference in evolutionary change in differential expression along branches that descend from speciation or duplication events (Figure 1b). Under this hypothesis, we would predict that contrasts across speciation nodes and duplication nodes are drawn from the same distribution (Figure XXEXPECTATIONS a). Under the alternative hypothesis specified by the ortholog conjecture, that there is a higher rate of change following duplication events than speciation events, we would expect to see the distribution of duplication contrasts shifted to higher values relative to the speciation contrasts (Figure XXEXPECTATIONS b).

```{r preliminaries, echo=FALSE, message=F, warning=F}
	library( treeio )
	library( tidyverse )
	library( magrittr )
	library( stringr )
	library( parallel )
	library( ggtree )
	library( ape )
	library( digest )
	library( hutan )
	library( gridExtra )

	# Set system computational parameters
	cores = detectCores() - 1
	if ( cores < 1 ) {
		cores = 1
	}
	set.seed( 23456 )
	
	# Name of the compara gene trees file. This is from 
	# ftp://ftp.ensembl.org/pub/release-75/emf/ensembl-compara/homologies/
	# but not included here due to its size. Put it in the kmrr/ directory
	gene_tree_name = "Compara.75.protein.nhx.emf"
	
	# Files produced by KMRR Rscript.R with Tau values and other expression statistics
	expression_file_names = c(
		"ChickenBrawandTScomparisonTable_9_6chPC.txt", 
		"ChimpBrawandTScomparisonTable_9_6cmPC.txt", 
		"GorillaBrawandTScomparisonTable_9_6gPC.txt", 
		"HumBrawandTScomparisonTable_9_6hPC.txt", 
		"MacacaBrawandTScomparisonTable_9_6mcPC.txt", 
		"MusBrawandTScomparisonTable_9_6mPC.txt", 
		"OpossumBrawandTScomparisonTable_9_6oPC.txt", 
		"PlatypusBrawandTScomparisonTable_9_6pPC.txt"
	)
	
	kmrr_directory = "kmrr/"
	expression_file_names %<>% paste( kmrr_directory, ., sep="" )
	gene_tree_name %<>% paste( kmrr_directory, ., sep="" )
	
	# The minimum number of genes with expression data that a gene needs to have to be considered
	min_genes_with_expression = 4
	
	# These clade dates are from the KMRR code, they got them from http://timetree.org/
	calibration_times = data.frame(
		age.min =
			c( 92, 167, 8, 296, 29, 162 ), 
		clade = 
			c( "Euarchontoglires", "Mammalia", "Homininae", 
					"Amniota", "Catarrhini", "Theria"
			),
		stringsAsFactors=FALSE
	)
```

```{r define_functions, echo=FALSE, message=F, warning=F}
	
	#' Get a boolean vector corresponding to all tips and internal nodes of 
	#' a tree, with value TRUE for tips and FALSE for internal nodes
	#' 
	#' @param nhx A phylogenetic tree as a treeio::treedata object
	#' @return A boolean vector
	#' @export
	is.tip.nhx = function( nhx ) {
		is.tip = rep( FALSE, nrow( nhx@data ) )
		is.tip[ 1:length( nhx@phylo$tip.label ) ] = TRUE
		is.tip
	}
	
	#' Record the distance from a tip for each node in a tree in 
	#' the data slot
	#' 
	#' @param nhx A phylogenetic tree as a  treeio::treedata
	#' @return A treeio::treedata object, with an additional node_age column
	#' in the data slot, or the original object if it was not of class treedata
	#' @export
	store_node_age = function( nhx ) {

		if ( class( nhx ) != "treedata" ) {
			return( nhx )
		}
		
		node_age = hutan::distance_from_tip( nhx@phylo )
		
		# make sure the dataframe is ordered by consecutive nodes
		stopifnot( all( nhx@data$node == 1:length( nhx@data$node ) ) )
		
		nhx@data$node_age = node_age 
		
		return( nhx )
	}
	
	
	#' Drop tips without expression data from a tree 
	#' 
	#' @param nhx A phylogenetic tree as a treeio::treedata object, with a 
	#' column @data$Tau containing expression data
	#' @param min_genes_with_expression The minimum number of tips with 
	#' expression data for a tree to be retained
	#' @return A treeio::treedata object, or NA if less than 
	#' min_genes_with_expression tips would have been retained
	#' @export
	drop_empty_tips = function( nhx, min_genes_with_expression ) {
		
		# Identify the tips without expression data
		to_drop = which( is.na( nhx@data[ 1:length( nhx@phylo$tip.label ), ]$Tau ) )
		
		# Return the pruned tree if it meets sampling criteria, otherwise return NA
		remaining = length( nhx@phylo$tip.label ) - length( to_drop )
		if( remaining >= min_genes_with_expression ){
			pruned = treeio::drop.tip( nhx, to_drop )
			return( pruned )
		}
		else{
			return( NA )
		}
	}


	#' Adjust branch lengths of a phylogenetic tree to make it ultrametric and 
	#' to time calibrate the speciation nodes to fixed values
	#' 
	#' @param nhx A phylogenetic tree as a treeio::treedata object, with speciation 
	#' nodes annotated with a value of "N" in @data$D
	#' @param calibration_times A dataframe with two columns: age.min is the age 
	#' of a clade, and clade is the name of the clade
	#' @param ... Any additional arguments to pass to ape::chronos()
	#' @return A treeio::treedata object if successfully calibrated
	#' @export
	calibrate_tree = function ( nhx, calibration_times, ... ) {
			
		# Create calibration matrix for speciation nodes
		calibration = 
			nhx@data[ !is.tip.nhx( nhx ), ] %>%
			filter( D == "N" ) %>%
			left_join( calibration_times, c( "label" = "clade" ) ) %>%
			select( node, age.min ) %>%
			mutate( age.max = age.min ) %>% 
			mutate( soft.bounds = NA )
		
		tree = try( 
			ape::chronos( nhx@phylo, calibration=calibration ) 
		)

		if( "phylo" %in% class( tree ) ){
			class( tree ) = "phylo"
			nhx@phylo = tree
			return( nhx )
		}
		else{
			return( NA )
		}
	}
	
	#' Calculate phylogenetic independent contrasts (PIC) with the tree and 
	#' character data in a treeio::treedata object
	#' 
	#' @param nhx A phylogenetic tree as a treeio::treedata object, with a 
	#' column @data$Tau containing expression data
	#' @return A treeio::treedata object, with new @data columns pic and var_exp
	#' @export
	pic.nhx = function( nhx ) {

		# Calculate the contrasts
		p = pic( nhx@data$Tau[ is.tip.nhx( nhx ) ], nhx@phylo, var.contrasts=TRUE )
	
		# Add the results back to the @data slot, padding the rows that correspond to 
		# tips with NA
		nhx@data$pic = c( rep( NA, length( nhx@phylo$tip.label ) ), p[ ,1 ] )
		nhx@data$var_exp = c( rep( NA, length( nhx@phylo$tip.label ) ), p[ ,2 ] )
	
		return( nhx )
	}

	#' Add phylogenetic independent contrasts (PIC) with the tree and 
	#' character data in a treeio::treedata object
	#' 
	#' @param gene_trees_calibrated A list of time calibrated phylogenetic trees 
	#' as treeio::treedata objects 
	#' @return A list of time phylogenetic trees with pics 
	#' as treeio::treedata objects
	#' @export
	add_pics_to_trees = function( gene_trees_calibrated ) {
	
		# Calculate independent contrasts for Tau on each tree, storing the results 
		# back into the @data slot of the tree objects
		gene_trees_pic = mclapply( gene_trees_calibrated, pic.nhx, mc.cores=cores )
		
		return( gene_trees_pic )
	}


	#' Add phylogenetic independent contrasts (PIC) with the tree and 
	#' character data in a treeio::treedata object
	#' 
	#' @param gene_trees_pic A list of phylogenetic trees with pics
	#' as treeio::treedata objects 
	#' @return A tibble with combined data, including phylogenetic independent 
	#' contrasts, for internal nodes of all trees
	#' @export
	summarize_contrasts = function( gene_trees_pic ) {
		# Combine the @data slots across all trees. These have labels, node annotations,
		# expression data, contrast results, etc...
		all_nodes = 
			lapply( 
				gene_trees_pic, 
				function( nhx ){
					tags = nhx@data
					tags$gene = digest( nhx ) # Creates a  hash that is unique to each gene tree
					tags %<>% select( -( B ) ) # B has inconsistent types, remove it
					return( tags )
				}
			) %>%
			bind_rows()
		
		# Make a new human readable column for plots that records speciation/ duplication
		all_nodes$Event = NA
		all_nodes$Event[ all_nodes$D == "Y" ] = "Duplication"
		all_nodes$Event[ all_nodes$D == "N" ] = "Speciation"
		
		# Create a subset that corresponds to nodes with contrasts. This excludes tips.
		# Also take the absolute value of the contrast, since we only consider magnitude.
		nodes_contrast = 
			all_nodes %>%
			filter( ! is.na( pic ) ) %>%
			mutate( pic = abs( pic ) ) %>%
			filter( ! is.na( D ) )
		
		return( nodes_contrast )
	}


	#' Calibrate a list of gene trees given a set of speciation node dates
	#' 
	#' @param gene_trees_sub A list of phylogenetic trees as treeio::treedata objects 
	#' with associated character data
	#' @param calibration_times A dataframe with two columns: age.min is the age 
	#' of a clade, and clade is the name of the clade
	#' @param ... Any additional arguments to pass to ape::chronos()
	#' @return A list of calibrated phylogenetic trees as treeio::treedata objects. May be 
	#' shorter than gene_trees_sub if some calibrations fail.
	#' @export
	calibrate_trees = function( gene_trees_sub, calibration_times, ... ) {
		# Make the trees ultrametric and calibrate the speciation nodes to 
		# specified times
		gene_trees_calibrated = 
			mclapply( 
				gene_trees_sub,
				calibrate_tree, 
				calibration_times=calibration_times,
				...,
				mc.cores=cores
			)
		
		# Remove trees that could not be successfully calibrated
		gene_trees_calibrated = gene_trees_calibrated[ ! is.na( gene_trees_calibrated ) ]
		
		# Parse the calibrated node ages from the internal @phylo object and store them with the 
		# corresponding rows in the @data object
		gene_trees_calibrated = 
			mclapply( 
				gene_trees_calibrated, 
				store_node_age, 
				mc.cores=cores 
			)
		
		return( gene_trees_calibrated )
	}


	#' Wilcoxon test of the ortholog conjecture, specifically that contrasts
	#' associated with duplication nodes are greater than those for 
	#' speciation nodes. A significant result indicates rejection of the null
	#' hypothesis that they are not greater.
	#' 
	#' @param nodes_contrast A tibble of independent contrasts
	#' @return Wilcoxon test p value
	#' @export
	wilcox_oc = function( nodes_contrast ) {
 
		p = wilcox.test( 
			nodes_contrast %>% filter( D=="Y" ) %>% .$pic, 
			nodes_contrast %>% filter( D=="N" ) %>% .$pic, 
			alternative="greater" 
		)$p.value
		
		return( p )
	}

```

```{r load_expression, echo=FALSE, message=F, warning=F}

	# Read the expression data for each species and combine into a single tibble
	expression = 
	lapply(
		expression_file_names, 
		read.table, 
		stringsAsFactors=FALSE, 
		header=TRUE
	) %>%
	bind_rows()

```

```{r load_trees, echo=FALSE, cache=TRUE, message=F, warning=F}

	# Read in the tree file as a vector of strings, one per line
	lines = readLines( gene_tree_name )

	# Only retain strings that are trees
	lines = lines[ grepl( "^\\(", lines, perl = TRUE ) ]

	# Parse each string to a treeio::treedata object, adding a label column 
	# for node names
	gene_trees = mclapply(
		lines,
		function( x ){
			tree_tc = textConnection( x )
			tree = treeio::read.nhx( tree_tc )
			close( tree_tc )
			tree@data$label = c( tree@phylo$tip.label, tree@phylo$node.label )
			return( tree )
		}, mc.cores=cores
	)
```


```{r add_expression_to_trees, echo=FALSE, cache=TRUE, message=F, warning=F}

	# Annotate each tree by joining the corresponding expression data to the 
	# @data object. This maps the expression data to the tree tips
	gene_trees_annotated = mclapply(
		gene_trees,
		function( tree ){
			tree@data %<>% left_join( expression, by = c( "G" = "Ensembl.Gene.ID" ) )
			return( tree )
		}, 
		mc.cores=cores
	)

	# Remove tips from the trees that do not have expression data
	gene_trees_sub = mclapply( 
		gene_trees_annotated, 
		drop_empty_tips, 
		min_genes_with_expression=min_genes_with_expression,
		mc.cores=cores 
	)

	# Remove trees with less than min_genes_with_expression tips with expression data
	gene_trees_sub = gene_trees_sub[ ! is.na( gene_trees_sub ) ]

```

```{r calibrate_trees, echo=FALSE, cache=TRUE, message=F, warning=F}

	# Time calibrate all speciation nodes to the same years
	gene_trees_calibrated = calibrate_trees( gene_trees_sub, calibration_times, model="correlated" )
	
```

```{r trees_to_contrasts, cache=TRUE, echo=FALSE, message=F, warning=F}

	# Calculate the contrasts from the trees and associated expression data
	gene_trees_pic = add_pics_to_trees( gene_trees_calibrated )

	# Collect all the contrasts in a single tibble
	nodes_contrast = summarize_contrasts( gene_trees_pic )
	
	# Test if the mean rank of duplication pics is greater than speciation pics
	wilcox_test_result = wilcox_oc( nodes_contrast )
	
```


```{r show_contrast_result, echo=FALSE, cache=TRUE, message=F, warning=F}

	# Plot the contrasts for duplication events and speciation events
	nodes_contrast %>%
		ggplot( aes( x=pic, y=..density.., col=Event ) ) + 
		geom_freqpoly( binwidth=.001, position="identity" ) +
		xlab( "Phylogenetic independent \ncontrasts (PIC)" ) +
		ylab( "Density" ) +
		xlim( 0, 0.075 ) + 
		theme_classic()

```

> Figure XXKMRR | Density plot of the magnitude of phylogenetic independent contrast values following duplication and speciation events. These contrasts are not larger following duplication events, as predicted by the ortholog conjecture.


When we reanalyze the data with phylogenetic independent contrasts (Fig. 1B), we did not find increased evolutionary change in expression following duplication events (Fig. XXKMRR). Of the `r length(gene_trees)` trees that were parsed from ENSMBL Compara [@Herrero:2016jj], `r length(gene_trees_sub)` passed taxon sampling criteria (`r min_genes_with_expression` genes) after removing tips without Tau values. Of these, `r length(gene_trees_calibrated)` were successfully time calibrated. These calibrated pruned trees were used to calculate phylogenetic independent contrasts for Tau. They contain `r nodes_contrast %>% filter(D=="Y") %>% nrow()` duplication nodes and `r nodes_contrast %>% filter(D=="N") %>% nrow()` speciation nodes.

Given the ortholog conjecture, we would expect to see larger contrasts following duplication events than speciation events. Instead, we find no such distribution (Figure XXKMRR). The Wilcox rank test does not reject the null hypothesis that the rate of evolution following duplications is the same as or less than the rate following speciation (p value = `r wilcox_test_result`). Our results therefore find no evidence for the ortholog conjecture in this system.

```{r ascertainment, cache=TRUE, echo=FALSE, fig.cap="XX_PIC_VAR", warning=FALSE, message=FALSE}

	# Examine by node age

	g_age =
		nodes_contrast %>%
		ggplot( aes( x=node_age, y=pic, col=Event ) ) + 
			geom_point( alpha=0.2 ) + 
			xlab( "Node age (millions of years)" ) + 
			ylab( "Phylogenetic independent contrasts (PIC)\n of differential expression" ) +
			ylim( 0, 0.3 ) +
			xlim( 0, 1000 ) +
			ggtitle( "(a)" ) +
			theme_classic()	

	max_speciation_age = 
		nodes_contrast %>% 
		filter( D=="N" ) %>%
		select( node_age ) %>%
		max( na.rm=TRUE )
	
	nodes_contrast_filtered_age = 
		nodes_contrast %>% 
		filter( node_age <= max_speciation_age )
	
	wilcox_filtered_age = wilcox_oc( nodes_contrast_filtered_age )

	# Examine by expected variance, ie branch length
	var_exp_max = 
		nodes_contrast %>% 
		filter( D=="N" ) %>%
		select( var_exp ) %>%
		max( na.rm=TRUE )

	var_exp_min = 
		nodes_contrast %>% 
		filter( D=="N" ) %>%
		select( var_exp ) %>%
		min( na.rm=TRUE )

	nodes_contrast_filtered_var = 
		nodes_contrast %>% 
		filter ( ( var_exp >= var_exp_min ) & ( var_exp <= var_exp_max ) )
	
	wilcox_filtered_var = wilcox_oc( nodes_contrast_filtered_var )

	
```

We next examined the possibility that ascertainment biases were differentially impacting the inference of evolution following duplication and speciation events. We focused on two possible sources of bias - node depth and branch length. We found no evidence that either affacted our results (Supplementary materials).

#### Understanding the incongruence between pairwise and phylogenetic methods

```{r pairwise, echo=FALSE, cache=TRUE, message=F, warning=F}
	get_pairwise_summary = function ( nhx ) {
		
		# Create a tibble with one row for each pairwise combination of tip nodes
		ntips = length( nhx@phylo$tip.label )
		D = 1:ntips %>% combn( 2 ) %>% t() %>% as_tibble()
		names( D ) = c( "tip_a", "tip_b" )
		
		# Add a colmn with the gene tree hash 
		D %<>% mutate( gene_tree = digest( nhx ) )
		
		# Add the tip names
		D %<>% mutate ( name_a = nhx@data$G[ D$tip_a ] )
		D %<>% mutate ( name_b = nhx@data$G[ D$tip_b ] )
		
		# Add the values at the tips
		D %<>% mutate ( tau_a = nhx@data$Tau[ D$tip_a ] )
		D %<>% mutate ( tau_b = nhx@data$Tau[ D$tip_b ] )
		D %<>% mutate ( difference = abs( tau_a - tau_b ) )
		
		# For each row, bet the most recent common ancestor of the pairwise tip combination
		D$mrca = 
			apply(
				D, 
				1, 
				function( y ) ape::getMRCA( 
					nhx@phylo, 
					c( as.integer( y['tip_a'] ), as.integer( y['tip_b'] ) )
				)
			) %>%
			as.integer()
		
		# Merge data on this mrca to the node
		D %<>% left_join(
			nhx@data %>%
				select( D, node, node_age, pic ),
			by = c( "mrca" = "node")
				
		)
		
		# Calculate the distance between the tips, which for an ultrametric tree 
		# is twice the age of their most recent common ancestor
		D %<>% mutate( distance = 2 * node_age )
		
		return( D )
		
	}

	# Build a tibble of all pairwise comparisons between tips
	pairwise_summary = 
		mclapply( gene_trees_pic, get_pairwise_summary, mc.cores=cores ) %>% 
		bind_rows()
	
	# Calcualte some summary statistics 
	ortholog_r = 
		pairwise_summary %>% 
		filter( D=="N" ) %>% 
		select( tau_a, tau_b ) %>% 
		cor( ) %>% 
		.[ 1, 2 ]
	
	paralog_r = 
		pairwise_summary %>% 
		filter( D=="Y" ) %>% 
		select( tau_a, tau_b ) %>% 
		cor( ) %>% 
		.[ 1, 2 ]
	
	paralog_subset_r = 
		pairwise_summary %>% 
		filter( D=="Y" ) %>% 
		filter( node_age <= max_speciation_age ) %>% 
		select( tau_a, tau_b ) %>% 
		cor( ) %>% 
		.[ 1, 2 ]
	
	ortholog_distance_mean = 
		pairwise_summary %>% 
		filter( D=="N" ) %>% 
		.[["distance"]] %>% 
		mean()
	
	paralog_distance_mean = 
		pairwise_summary %>% 
		filter( D=="Y" ) %>% 
		.[["distance"]] %>% 
		mean()
	
```

In order to better understand why our phylogenetic analysis seems to support a different conclusion (*i.e.*, no support for the ortholog conjecture) than the published analysis of KMRR [@KryuchkovaMostacci:2016iw] (*i.e.*, strong support for the ortholog conjecture), we first checked to make sure we could reproduce their result based on pairwise analyses. This is important since we are only looking at at a subset of the data they considered, the Brawand *et al.* 2011 [@Brawand:2011du] dataset for gene trees that could be successfully time calibrated. We can replicate their result - we find a higher Tau correlation between orthologs (R=`r round( ortholog_r, 3) `) then between paralogs (R=`r round( paralog_r, 3)`).

Why is it that pairwise methods and phylogenetic methods lead to opposite conclusions? The reason is that they describe different things. Pairwise comparisons describe contemporary patterns, while phylogenetic methods infer historical processes. There need not be a different process of evolution following speciation and duplication for paralogs to be more different than orthologs. This could be due to the structure of the gene phylogenies. If paralogs tend to be more distantly related to each other than orthologs, then there would be more time for differences to accumulate even if the rate of change is the same between the two. This is, in fact, the case for these data. While the mean distance between orthologs is `r round( ortholog_distance_mean, 1 )` years, the mean distance between paralogs is `r round( paralog_distance_mean, 1 )` years. As described above, this is because the oldest speciation event is by definition the most recent common ancestor of the the species included in the study, but many gene families underwent duplication before this time.

To test the hypothesis that distantly related paralogs arising from these older duplication events are the reason for the lower correlation of Tau between paralogs than between orthologs, we removed them. When we consider only the paralogs that arise from duplication events the same age or younger than the oldest speciation event, the observed pattern is very different. The correlation of Tau between these shallower paralog pairs, R=`r round( paralog_subset_r, 3)`, is much stronger than for the correlation between all paralog pairs, R=`r round( paralog_r, 3)`. This is much closer to the ortholog correlation, which has R=`r round( ortholog_r, 3) `.

The reason that paralog pairs have Tau values that are more different than ortholog pairs is that they tend to be more distantly related, not because there are different rates of Tau evolution following duplication and speciation events. This means that having information on whether two genes are orthologs or paralogs provides little added information about expression beyond knowing how distantly related the two genes are. This has several implications. It is an example of the limitations of the concepts of orthology and paralogy [@Dunn:2016er]. These terms can have straightforward meaning in small gene trees with simple duplication/speciation histories, but the utility of the terms breaks down on larger more complex gene trees. Orthology and paralogy are annotations on the tips of the tree that are derived from the inferred structure of the tree and annotations of duplication and speciation at internal tree nodes. In this sense, orthology and paralogy are statements about the internals of the tree that are distilled into statements at the tips of the tree. Much is lost in the process, though. In this case, KMRR [@KryuchkovaMostacci:2016iw] relied on orthology and paralogy annotations that are derived from the inferred duplication and speciation history on gene phylogenies [@Herrero:2016jj]. These are the same pre-computed trees we use in our phylogenetic re-analyses of their study. For most questions it is much more direct to focus on the structure of the tree and the inferred processes within the tree, such as which internal nodes are duplication or speciation events and how much change occurs along the branches.

#### Implications for the ortholog conjecture

Our reanalysis of KMRR using phylogenetic methods suggests that the ortholog conjectures is not a dominant pattern that is central to explaining the evolution of phenotypic diversity in gene families. This has important biological implications. It suggests that the mechanism of gene divergence (speciation versus duplication) may not have as strong an impact on phenotypic divergence as thought. It also has technical implications. Rather than focus on whether genes are orthologs or paralogs when attempting to predict function, it may be more effective to simply focus on how closely related or distantly related they are. Closely related paralogs, for example, may tend to have more similar phenotypes than more distantly related orthologs. 

The widespread invocation of the ortholog conjecture follows from the expectation that it is strong, widely distributed phenomenon. Our failure to detect a signature consistent with the ortholog conjecture suggests that it may not be, and that it should not be assumed in systems where it has not been explicitly tested. 

### Levin *et al.* reanalysis

#### Original pairwise analyses of developmental gene expression

![Figure Levin](levin_etal/Figure_levin.png?raw=true)

> Figure XXLevin | Distributions of pairwise similarity scores for each phase of development. Pairwise scores for the ctenophore are red. Wilcoxon test p-values for the significance of the differences between early-mid distributions and late-mid distributions are on the right. Model of variance, which is inversely related to similarity, is on the left. (a) The distributions as published. Low similarity (*i.e.*, high variance) in the mid phase of development was interpreted as support for an inverse hourglass model for the evolution of gene expression. The five least-similar mid phase scores were all from the ctenophore. Published KS p-values, based on duplicated data, are in parentheses. The inset ctenophore image is by S. Haddock from phylopic.org. (b) The distributions after the exclusion of the ctenophore. The early and mid phase distributions are not statistically distinct. This suggest a wine bottle model, with similar evolutionary variance at the early and mid phase and less at the late phase.

How changes in animal development relate to the evolution of animal diversity is a major question in evolutionary developmental biology (EvoDevo). To address this topic, Levin *et al.* [@Levin:2016bp] analyzed gene expression through the course of embryonic development for ten animal species, each from a different phylum. As their title ("The mid-developmental transition and the evolution of animal body plans") indicates, they arrived at two major conclusions. First, animal development is characterized by a well-defined mid-developmental transition that marks the transition from an early phase of gene expression to a late stage of gene expression. Second, this transition helps explain the evolution of features observed among distantly related animals. Specifically, they concluded that animals from different phyla exhibit an "inverse hourglass" model for the evolution of gene expression, where there is more evolutionary variance in gene expression at a mid phase of development than there is at early and late phases. Closely related animals have previously been described as having an hourglass model of gene expression, where evolutionary variance in expression is greater early and late in development than at the midpoint of development [@Kalinka:2010dw; @DomazetLoso:2010cn]. Levin *et al.* conclude that this contrast between distantly and closely related animals provides biological justification for the concept of phyla and may provide a definition of phyla. 

Levin *et al.* [@Levin:2016bp] arrived at this conclusion by making pairwise comparisons of ortholog expression through development between species. They characterized each gene in each species as having expression that peaks in early, mid, or late temporal phase of development. The goodness of fit to these patterns was not considered, and alternative patterns, such as the possibility that there are more phases of expression, were not evaluated. For each species pair, they identified the orthologs shared by these species. This list of shared genes was different from species pair to species pair. They then calculated a similarity score for each temporal phase for each species pair based on the fraction of genes that exhibited the same patterns in each species. The distributions of similarity scores are plotted in their [Figure 4d](http://www.nature.com/nature/journal/v531/n7596/fig_tab/nature16994_F4.html), and their Kolmogorovâ€“Smirnov (KS) tests indicated that the early distribution and late distribution were each significantly different from mid distribution (P < 10<sup>-6</sup> and P < 10<sup>-12</sup>, respectively). This is the support they presented for the inverse hourglass model.


#### Reexamination of pairwise comparisons

We previously described some concerns with their interpretations of these results [@Hejnol:2016hw]. The main concern was that because they only sampled one species per phylum, there is no way to tell if observed differences are unique to the branches that gave rise to clades designated as phyla or occurred along the many other branches in the tree. Here we examine the support for the inverse hourglass model itself. We focus on the matrix of pairwise comparisons that their KS tests and their Figure 4d are based on. 

The first problems we identified were general stastistical issues that are not problems with pairwise comparisons *per se*. First, in the published analyses every data point was included twice because both reciprocal comparisons (which have the same values) were retained. For example, there is both a nematode to arthropod comparison and an arthropod to nematode comparison. As a consequence, there are 90 entries for the 45 pairwise comparisons, and by doubling the data the significance of the result appears stronger than it actually is. After removing the duplicate values, the p values are far less significant, 0.002 for the early-mid comparison and on the order of 10<sup>-6</sup> for early-late. Second, the test they used (KS test) is not appropriate for the hypothesis they seek to evaluate. The KS test does not just evaluate whether one distribution is greater than the other, it also tests whether the shape of the distributions are the same. In addition, the samples in this dataset are matched (*i.e.*, for each pairwise comparison there is a early, mid, and late expression value), which the KS test does not take into account. The Wilcoxon test is instead appropriate in this case. When applied to the de-duplicated data, the significance of this test is 0.02 for the early-mid comparison and on the order of 10<sup>-7</sup> for early-late.

The next issue we found is a general problem with pairwise comparisons. We found that all five of the lowest values in the mid phase distribution (Figure XXLevin a) are for pairwise comparisons that include the ctenophore (comb jelly). When the nine pairwise comparisons that include the ctenophore are removed and the general statistical issues noted above are addressed, there is no significant difference between the early phase and mid phase distributions (P = 0.1428 for the early-mid comparison and P < 10<sup>-5</sup> for the late-mid comparison). The inverse hourglass turns into a bottle (Figure XXLevin b). 

This highlights a well understood property of pairwise comparisons across species [@Felsenstein:1985ua]: evolutionary changes along a given branch, like those along the ctenophore branch, impact each of the multiple pairwise comparisons that includes that branch. The pairwise comparisons are therefore not independent - different pairwise comparisons are impacted by canges along some of the same branches. This can give the impression of a general pattern across the tree that is instead specific to changes along one part of the tree. The number of comparisons impacted by each change depends on the structure of the phylogenetic tree, *i.e.* how the species are related to each other.

Phylogenetic comparative methods were developed specifically to address this problem [@Felsenstein:1985ua]. While we do demonstrate this problem impacts the Levin *et al.* analysis, we did not perform a phylogenetic reanalysis of this study, as we did for the KMRR study. This is because the similarity metric computed in the pairwise comparisons of Levin *et al.* are based on different genes for different species pairs. This means that they do not generate a trait that can be evaluated across a phylogeny, as KMRR's Tau can be. A full phylogenetic reanalysis would be possible by using upstream analysis products to re-derive new expression summary statistics suitable for phylogenetic comparative methods.


### Phylogenetic comparative methods in functional genomics

Our results highlight the importance of explicitly accounting for phylogenetic relationships when studying character evolution, including developmental [@Telford:2003tsa] and functional genomic traits [@Hejnol:2015gr]. This is particularly true for evolutionary analyses of quantitative gene expression [@Dunn:2013cpa]. 

Some of the most widely used phylogenetic comparative methods [@Felsenstein:1985ua; @Grafen:1989um] are already directly applicable to CFG data. There are interesting new challenges at this interdisciplinary interface that will need to be addressed to fully realize the potential of phylogenetic comparative functional genomics studies. One such challenge is that most phylogenetic comparative analyses of covariance between traits have been developed to address problems with many more species (e.g., dozens or more) relative to the number of traits being examined. In CFG analyses, there are often far fewer species because adding taxa is still expensive, but tens of thousands of traits. This creates statistical challenges as the resulting covariance matrices are singular and, if not treated appropriately, imply many false correlations that are artifacts of project design. We outlined these challenges and potential solutions in the context of gene expression [@Dunn:2013cpa]. In addition to methods development, there is a critical need for software tools that make it simpler to integrate existing functional genomic and phylogenetic comparative packages.

Recent advances in phylogenetic comparative methods are particularly well suited to addressing questions about the evolution of functional genomic traits. Most early phylogenetic comparative methods attempted to remove evolutionary signal to correct statistical tests for correlations between traits, while more recent methods tend to focus on testing hypotheses of evolutionary processes [@Pennell:2013cw]. The application of this newer focus to functional genomics provides an exciting opportunity to address long standing questions of broad interest, including the order of changes in functional genomics traits and shifts in rates of evolution of one functional genomics trait following changes in another trait. 

We are not the first to apply phylogenetic comparative methods to functional genomic data. While the vast majority of comparative functional genomics studies have used pairwise similarity methods, a small number of CFG studies have employed phylogenetic comparative approaches [@Oakley:2005ky; @Eng:2009gp; @Chang:2014hq]. For instance, a phylogenetic ANOVA [@Rohlfs:2015bd] of the evolution of gene expression improves statistical power and drastically reduces the rate of false positives relative to pairwise approaches. 

Addressing the statistical non-independence of pairwise comparisons is not the only advantage of using phylogenetic comparative methods for functional genomics analyses. Another problem with the pairwise comparisons is that, except at the tips, they summarize changes along many branches in the phylogeny. Two paralogs that diverged from a duplication event deep in the tree may have many subsequent duplication and speciation events, and changes along all these branches will impact the final pairwise comparison. Phylogenetic methods have the advantage of isolating the changes under consideration (Fig. 1b). The phylogenetic methods avoid diluting the change that occurs along the branches that follow the node in question with changes along all subsequent branches. There may still be missing speciation events, due to extinction and incomplete taxon sampling, and missing duplication events, due to gene loss, but these omissions affect both methods.

## Conclusions

The fact that the first two CFG studies we reanalyzed show serious problems with pairwise comparisons indicates that there likely to be similar problems in other CFG studies, and that future CFG studies will be compromised if they continue to use pairwise methods. Studies of evolutionary functional genomics should not be focused on the tips of the tree, they should explicitly delve into the tree itself.

## Methods

All files needed to re-execute the analyses presented in this document are available at https://github.com/caseywdunn/comparative_expression_2017. The most recent commit at the time of the analysis presented here was `r system("git log | head -n 1", intern=TRUE) %>% str_replace("commit ", "")`.


### KMRR reanalysis

The KMRR study [@KryuchkovaMostacci:2016iw] followed excellent practices in reproducibility. They posted all data and code needed to re-execute their analyses at figshare: https://figshare.com/articles/Tissue-specificity_of_gene_expression_diverges_slowly_between_orthologs_and_rapidly_between_paralogs/3493010/2 . We slightly altered their `Rscript.R` to simplify file paths. This modified script and their data files are available in the github repository for this paper, as are the intermediate files that were generated by their analysis script that we used in our own analyses. We also obtained the `Compara.75.protein.nh.emf` gene trees [@Herrero:2016jj] from ftp://ftp.ensembl.org/pub/release-75/emf/ensembl-compara/homologies/ and place them in the same directory as this file.

We considered only the data from Brawand *et al.* 2011 [@Brawand:2011du] for the eight taxa included in KMRR. We left in sex chromosome genes and testes expression data, as this corresponded to the analyses that provided the strongest support for the ortholog conjecture and therefore the most conservative reconsideration of it.

Each of these gene trees includes includes homologous sequences derived from a common ancestral sequence. The tip nodes of a gene tree can include both orthologs and paralogs, and each internal node is an inferred speciation or duplication event. Prior to phylogenetic analyses of gene expression, each tree was first pruned to include only tips for which expression data were available. The speciation nodes in the tree were then time calibrated so that branch lengths were consistent across trees. We then calculated the magnitude of the independent contrast for differential expression at each internal node in each of these trees.

### Levin et al. reanalysis

Levin *et al.* helpfully provided data and clarification on methods. We obtained the matrix of pairwise scores that underlies their [Figure 4d](http://www.nature.com/nature/journal/v531/n7596/fig_tab/nature16994_F4.html) and confirmed we could reproduce their published results. We then removed duplicate rows, identified ctenophores as overrepresented among the low outliers in the mid-developmental transition column, and applied the Wilcoxon test in place of the Kolmogorov-Smirnov test. These data, our analysis code, and additional information are available in a git repository at [https://github.com/caseywdunn/levin2016](https://github.com/caseywdunn/levin2016). This repository includes our analysis notebook at [https://rawgit.com/caseywdunn/levin2016/master/reanalyses.html](https://rawgit.com/caseywdunn/levin2016/master/reanalyses.html).

## Acknowledgments

Thanks to XX. This work was supported by the National Science Foundation (DEB-1256695 and the Waterman Award).


## Supplementary Material

### KMRR Analyses

#### Investigation of potential ascertainment bias

While the age of speciation nodes is constrained, duplication nodes can be much older and therefore have a wider range of ages (Supplementary Figure XXAscertainment a). This is because many gene duplication events are older than the most recent common ancestor of the species in the study. There are also technical factors that can lead to an excess of duplication events deeper in the tree. Gene tree estimation errors, for example, tend to lead to the overestimation of deep duplications [@Hahn:2007cj]. If independent contrast values also tended to to be lower at greater node depth, it could interact with the preponderance of duplications at greater depth to create a pattern of lower contrasts associated with duplication events. To test for such an effect, we remove duplication nodes that are older than the oldest speciation node. Our results are unchanged and this reduced dataset does not reject the null hypothesis that the rate of evolution following duplications is the same as or less than the rate following speciation (Supplementary Figure XXAscertainment b).

The independent contrast across a node is the amount of change observed between the daughter notes, scaled by the expected variance [@Felsenstein:1985ua]. The expected variance is principally determined by the lengths of the branches leading from the node to these daughters. The shorter the total length of the two branches leading to daughter nodes, the larger the contrast for a given observed difference. This is because the same difference across a shorter total branch length indicates a greater rate of evolutionary change. The expected variance of contrasts for speciation nodes is constrained by the branch lengths on the species tree, but the expected variance of contrasts for duplications has a much wider range (Figure XXAscertainment c). This could lead to biases if the lengths of branches that descend from duplication nodes tend to be overestimated. We therefore examined only the contrasts that fell within in the range of expected variance seen for speciation contrasts, excluding duplication contrasts that fall outside of this range. This reanalysis does not reject the null hypothesis either (Figure XXAscertainment d), indicating that branch length bias is not responsible for rejecting the ortholog conjecture.

```{r ascertainment_plot, cache=TRUE, echo=FALSE, fig.cap="XX_PIC_VAR", warning=FALSE, message=FALSE}

	# Plot results of the ascertainment bias analyses

	label_age = str_c( "Wilcoxon p=", wilcox_filtered_age )
	
	g_density_filtered_age =
		nodes_contrast_filtered_age %>%
		ggplot( aes( x=pic, y=..density.., col=Event ) ) + 
		geom_freqpoly( binwidth=.001, position="identity" ) +
		xlab( "Phylogenetic independent \ncontrasts (PIC)" ) +
		ylab( "Density" ) +
		xlim( 0, 0.075 ) + 
		ggtitle( "(b)" ) +
		annotate( "text", x = Inf, y = Inf, hjust="right", vjust="top", label=label_age ) +
		theme_classic()


	# Examine by expected variance (branch length)
	g_var_exp =
		nodes_contrast %>%
		ggplot( aes( x=var_exp, y=pic, col=Event ) ) + 
		geom_point( alpha=0.2 ) + 
		xlab( "Expected variance, \n depends mostly on branch length" ) + 
		ylab( "Phylogenetic independent contrasts (PIC)\n of differential expression" ) +
		ylim( 0, 0.3 ) +
		xlim( 0, 2000 ) +
		ggtitle( "(c)" ) +
		theme_classic()	

	label_var = str_c( "Wilcoxon p=", wilcox_filtered_var )
	
	g_density_filtered_var =
		nodes_contrast_filtered_var %>% 
		ggplot( aes( x=pic, y=..density.., col=Event ) ) + 
		geom_freqpoly( binwidth=.001, position="identity" ) + 
		ggtitle( "(d)" ) +
		theme( legend.justification=c( 1, 1 ), legend.position=c( 1, 1 ) ) +
		xlim( 0, 0.075 ) + 
		theme_classic() +
		annotate( "text", x = Inf, y = Inf, hjust="right", vjust="top", label=label_var ) +
		xlab( "Phylogenetic independent contrasts (PIC)\n of differential expression" ) +
		ylab( "Density" )

	grid.arrange(
		g_age,
		g_density_filtered_age,
		g_var_exp,
		g_density_filtered_var,
		ncol = 2
	)
```

> Supplementary Figure XXAscertainment | Investigation of possible ascertainment biases. (a) Magnitude of independent contrasts plotted against node age. Speciation nodes are calibrated to particular times, whereas duplication nodes have a wider range. (b) Density plot of contrasts for only the nodes that have an age less than or equal to the maximum age of speciation nodes. (c) Magnitude of independent contrasts plotted against expected variance, which is largely determined by branch lengths. Contrasts for speciation nodes have a narrower range of expected variance than do contrasts for duplication nodes. (d) Density plot of contrasts for only the nodes that have expected variance within the range of contrasts across speciation nodes.


## References
