---
bibliography: manuscript.bib
csl: nature.csl
output:
  github_document: default
  pdf_document:
    dev: png
---

```{r setup, cache=TRUE, message=F, echo=FALSE, warning=F}
knitr::opts_chunk$set(echo = TRUE)
```

# The importance of phylogenetic methods when comparing functional genomic data across species

Casey W. Dunn^1^*, Felipe Zapata^1,2^, Cat Munro^1^, Stefan Siebert^1,3^, Andreas Hejnol^4^

^1^ Department of Ecology and Evolutionary Biology, Brown University, Providence, RI, USA

^2^ Current address: Department of Ecology and Evolutionary Biology, University of California Los Angeles, Los Angeles, CA, USA

^3^ Current address: Department of Molecular & Cellular Biology, University of California at Davis, Davis, CA, USA

^4^ Sars Institute, Bergen, Norway


\* Corresponding author, casey_dunn@brown.edu


## Abstract

Functional genomics, the study of how genomes work, is one of the most rapidly advancing fields in Biology. There is now considerable interest in comparing genome function across species to understand what is conserved, what is variable, and how the diversity of organism traits (including anatomy, physiology, and development) relates to genome diversity. Most published comparisons of genome function across species have, however, neglected to use methods that consider the evolutionary relationships among species. This has two potential negative impacts. First, it can lead to the wrong conclusions about the evolution of genome function. Second, it is a missed opportunity to learn about biology that can only be understood in an explicit historical context that considers these relationships. Here we examine two recently published comparative expression studies that used multiple pairwise comparison rather than phylogenetic approaches to investigate evolutionary patterns of gene expression. We find problems with the pairwise comparisons used in both studies that undermine their published conclusions. This is a concrete demonstration of the inadequacy of pairwise comparisons in comparative genomics studies, and indicates that it will be critical to adopt phylogenetic comparative methods in future analyses.

## Introduction

The focus of genomics research has quickly shifted from describing genome sequences to Functional Genomics (FG), the study of how genomes "work" using tools that measure functional attributes such as expression, chromatin state, and transcription initiation. FG, in turn, is now becoming more comparative- there is great interest in understanding how FG variation across species gives rise to a diversity of development, morphology, physiology, and other phenotypes [@Wray:2013gh]. Comparative Functional Genomics (CFG) analyses are also critical to transferring functional insight across species, and will grow in importance in coming years.

A rich theoretical and statistical methodology of phylogenetic comparative methods have been developed over the last three decades to address the challenges and opportunities of comparisons across species [@Felsenstein:1985ua; @Grafen:1989um; @Chamberlain:2012vx; @Garamszegi:2014ew; @Pagel:1999gc]. These have largely been applied to the evolution of morphological and ecological traits, but are just as relevant to CFG. Most CFG studies have abstained from phylogenetic approaches and instead heavily rely on pairwise comparisons (Fig. 1A) that do not account for the fact that evolutionary relationships explain much of the structure of variation across species. This leaves CFG studies susceptible to serious statistical errors and is a missed opportunity to ask questions that are only accessible in an explicit phylogenetic context.

One reason that CFG has not yet embraced phylogenetic approaches is that it has not yet been concretely demonstrated that pairwise and phylogenetic approaches can lead to different results in CFG. Although the value of phylogenetic approaches has been repeatedly shown in comparative analyses of other character data [@Chamberlain:2012vx; @Ricklefs:1996wb], this value is apparently not well known to the FG community since the field has until recently largely focused on within-species analyses. Many initial CFG studies have also been focused on only two species at a time, where the differences between phylogenetic and pairwise approaches are not as pronounced. As the number of species considered in each CFG study grows, the shortcomings of pairwise approaches will become far more acute. 

Most CFG studies to date compare gene expression across species, often between distinct tissues or developmental time points. Gene expression has been a popular target of CFG studies because, with the advent of RNA-Seq, it is the most readily collected FG trait across diverse species. Here we reconsider two recent comparative analyses of gene expression data [@KryuchkovaMostacci:2016iw; @Levin:2016bp]. 

Kryuchkova-Mostacci and Robinson-Rechavi (KMRR) 2016 [@KryuchkovaMostacci:2016iw] integrated vertebrate expression datasets to test the ortholog conjecture - the hypothesis that orthologs tend of have more conserved attributes (expression in this case) than do paralogs [@Nehrt:2011bm]. Using pairwise comparisons (Fig. 1A), they found lower expression covariance between paralogs than between orthologs and interpreted this as strong support for the ortholog conjecture. Orthology and paralogy is annotated based on gene phylogenies, but these phylogenies are not used when making the expression comparisons.

The second CFG study we evaluate here is Levin *et al.* [@Levin:2016bp]. This study analyzed gene expression through the course of embryonic development for ten animal species, each from a different phylum. They concluded that animals from different phyla exhibit an "inverse hourglass" model for the evolution of gene expression, where there is more evolutionary variance in gene expression at a mid phase of development than there is at early and late phases (Fig. 2). This contrasts with the "hourglass" model previously proposed for closely related species [@Kalinka:2010dw]. Levin *et al.* conclude that this provides biological justification for the concept of phyla and may provide a definition of phyla. We previously described concerns with their interpretations of these results [@Hejnol:2016hw]. Here we directly address the analyses themselves by examining the structure of the pairwise comparisons. 

![Figure Overview](Figure_overview.png?raw=true)

> Figure 1 | Pairwise and phylogenetic comparative approaches. (a) Many comparative functional genomic studies across species rely on pairwise comparisons, where traits of each gene are compared to traits of every other gene. This leads to many more comparisons than unique observations, making each comparison non-independent. (b) Comparative phylogenetic methods, including phylogenetic independent contrasts [@Felsenstein:1985ua], make a smaller number of independent comparisons, where each contrast measures changes along different branches. Phylogenetic approaches are rarely used for functional genomics studies. 


## Results

### KMRR Reanalysis

The KMRR considers several publicly available datasets. Here we consider their conclusions in the context of Brawand *et al.* 2011 [Brawand:2011du], one of the studies they considered. This analysis of XX different organs across XX mammal species.

For each internal node in each gene tree (Figure 1b), we then measured the phylogenetically independent contrast (PIC) in the expression across this node [@Felsenstein:1985ua]. This is the difference in values of descendent nodes (i.e., the evulutionary change along the branches that descend from the node) scaled by the expected variance (determined largely by the lengths of these branches). These contrasts were then annotated by whether they are made across branches that descend from a speciation or duplication event. The original description of independent contrasts [@Felsenstein:1985ua] focused on assessing covariance between changes in two traits on a tree. Our use of it to look for differences in evolutionary changes of one trait (differential expression) between two categories of nodes (speciation and duplication) on a tree is a bit different than the usual application of the method.


![Figure Expectations](figure_expectations.png?raw=true)

> Figure XXExpectations | (a) Null hypothesis. (b) Under the ortholog conjecture.


To test the ortholog conjecture it is essental to have a null hypothesis that results in a distinct prediction from the ortholog conjecture. A suitable null hypothesis is that there is no difference in evolutionary change in differential expression along branches that descend from speciation or duplication events (Figure 1b). Under this hypothesis, we would predict that contrasts across speciation nodes and duplication nodes are drawn from the same distribution (Figure XXEXPECTATIONS a). Under the alternative hypothesis specified by the ortholog conjecture, that there is a higher rate of change following duplication events than speciation events, we would expect to see the distribution of duplication contrasts shifted to higher values relative to the speciation contrasts (Figure XXEXPECTATIONS b).

When we reanalyze the data with phylogenetic independent contrasts (Fig. 1B), we did not find increased evolutionary change in expression following duplication events (Fig. XX).

```{r preliminaries, echo=FALSE, message=F, warning=F}
	library( treeio )
	library( tidyverse )
	library( magrittr )
	library( stringr )
	library( parallel )
	library( ggtree )
	library( ape )
	library( digest )
	library( hutan )
	library( gridExtra )

	# Set system computational parameters
	cores = detectCores() - 1
	if ( cores < 1 ) {
		cores = 1
	}
	set.seed( 23456 )
	
	# Set biology starting parameters, including files

	gene_tree_name = "Compara.75.protein.nhx.emf"
	
	kmrr_directory = "kmrr/"
	
	expression_file_names = c(
		"ChickenBrawandTScomparisonTable_9_6chPC.txt", 
		"ChimpBrawandTScomparisonTable_9_6cmPC.txt", 
		"GorillaBrawandTScomparisonTable_9_6gPC.txt", 
		"HumBrawandTScomparisonTable_9_6hPC.txt", 
		"MacacaBrawandTScomparisonTable_9_6mcPC.txt", 
		"MusBrawandTScomparisonTable_9_6mPC.txt", 
		"OpossumBrawandTScomparisonTable_9_6oPC.txt", 
		"PlatypusBrawandTScomparisonTable_9_6pPC.txt"
	)
	
	expression_file_names %<>% paste(kmrr_directory, ., sep="")
	
	gene_tree_name %<>% paste(kmrr_directory, ., sep="")
	
	
	# These are from the KMRR code
	calibration_times = data.frame(
		age.min =
			c(20, 92, 167, 8, 42, 74, 296, 535, 104, 937, 29, 722, 441, 65, 15, 1215, 
				414, 371, 162, 25, 74, 77, 86), 
		clade = c("Hominoidea", "Euarchontoglires", "Mammalia", "Homininae", 
							"Simiiformes", "Primates", "Amniota", "Vertebrata", "Eutheria", 
							"Bilateria", "Catarrhini", "Chordata", "Euteleostomi", 
							"Haplorrhini", "Hominidae", "Opisthokonta", "Sarcopterygii", 
							"Tetrapoda", "Theria", "Murinae", "Sciurognathi", "Rodentia", "Glires"),
		stringsAsFactors=FALSE
	)
	
	
	
	# Functions
	
	# Returns a logical vector indicating which rows in nhx_tags are for tips
	is.tip.nhx = function( nhx ) {
	  is.tip = rep( FALSE, nrow( nhx@data ) )
	  is.tip[ 1:length( nhx@phylo$tip.label ) ] = TRUE
	  is.tip
	}
	
	# put node ages in @data
	store_node_age = function( nhx ) {
  
	  if ( class( nhx ) != "treedata" ) {
	    return( nhx )
	  }
	  
	  node_age = hutan::distance_from_tip( nhx@phylo )
	  
	  # make sure the dataframe is ordered by consecutive nodes
	  stopifnot( all( nhx@data$node == 1:length( nhx@data$node ) ) )
	  
	  nhx@data$node_age = node_age 
	  
	  return( nhx )
	}
	
	# Make tree ultrametric, calibrating speciation nodes to consistent times
	calibrate_tree = function ( nhx, calibration_times ) {
			
			# Create calibration matrix for speciation nodes
			calibration = 
				nhx@data[ !is.tip.nhx( nhx ) ,  ] %>%
				filter( D=="N" ) %>%
				left_join( calibration_times, c( "label" = "clade" ) ) %>%
				select( node, age.min ) %>%
				mutate( age.max = age.min ) %>% 
				mutate( soft.bounds = NA )
			
			tree = try( ape::chronos( nhx@phylo, calibration=calibration, model="correlated" ) )
			if( "phylo" %in% class(tree) ){
				class( tree ) = "phylo"
				nhx@phylo = tree
				return( nhx )
			}
			else{
				return( NA )
			}
		}

```

```{r load_expression, echo=FALSE, message=F, warning=F}
	expression = lapply(
		expression_file_names, read.table, stringsAsFactors=FALSE, header=TRUE
	) %>%
	bind_rows()
```

```{r load_trees, echo=FALSE, cache=TRUE, message=F, warning=F}
	lines = readLines( gene_tree_name )

	# Only consider the lines with trees
	lines = lines[ grepl( "^\\(", lines, perl = TRUE ) ]

	gene_trees = mclapply(
		lines,
		function( x ){
			tree_tc = textConnection( x )
			tree <- treeio::read.nhx( tree_tc )
			close( tree_tc )
			tree@data$label = c( tree@phylo$tip.label, tree@phylo$node.label )
			return( tree )
		}, mc.cores=cores
	)
```


```{r add_expression_to_trees, echo=FALSE, cache=TRUE, message=F, warning=F}
	gene_trees_annotated = mclapply(
		gene_trees,
		function( tree ){
			tree@data %<>% left_join( expression, by = c("G" = "Ensembl.Gene.ID") )
			return( tree )
		}, mc.cores=cores
	)
```

```{r prep_trees, echo=FALSE, cache=TRUE, message=F, warning=F}
	drop_empty_tips = function( nhx ) {
	  to_drop = which( is.na( nhx@data[ 1:length(nhx@phylo$tip.label), ]$Tau ) )
	  remaining = length(nhx@phylo$tip.label) - length(to_drop)
	  
	  if( remaining > 3 ){
	  	pruned = treeio::drop.tip( nhx, to_drop )
	  	return( pruned )
	  }
	  else{
	  	return( NA )
	  }
	}

	gene_trees_sub = mclapply( gene_trees_annotated, drop_empty_tips, mc.cores=cores )
	gene_trees_sub = gene_trees_sub[ ! is.na( gene_trees_sub ) ]
	
	
	gene_trees_calibrated = mclapply( 
		gene_trees_sub,
		calibrate_tree, 
		calibration_times=calibration_times,
		mc.cores=cores
	)
	
	gene_trees_calibrated = gene_trees_calibrated[ ! is.na( gene_trees_calibrated ) ]
	
	gene_trees_calibrated = mclapply( gene_trees_calibrated, store_node_age, mc.cores=cores )
	
```

Of the `r length(gene_trees)` trees that were parsed, `r length(gene_trees_sub)` passed taxon sampling criteria and `r length(gene_trees_calibrated)` were successfully time calibrated.

```{r contrasts, cache=TRUE, echo=FALSE, message=F, warning=F}

	pic.nhx = function( nhx ) {
  
	  p = pic( nhx@data$Tau[ is.tip.nhx( nhx ) ], nhx@phylo, var.contrasts=TRUE )
	  
	  nhx@data$pic = c( rep( NA, length( nhx@phylo$tip.label ) ), p[ ,1 ] )
	  nhx@data$var_exp = c( rep( NA, length( nhx@phylo$tip.label ) ), p[ ,2 ] )
	  
	  return( nhx )
	}

	gene_trees_pic = mclapply( gene_trees_calibrated,  pic.nhx, mc.cores=cores )
	
	all_nodes = lapply( 
	      gene_trees_pic, 
	  		function( nhx ){
	  			tags = nhx@data
	  			tags$gene = digest( nhx )
	  			tags %<>% select( -(B) ) # B has inconsistent types, remove it
	  			return(tags)
	  		}
  		) %>%
		bind_rows()
	
	all_nodes$Event = NA
	
	all_nodes$Event[ all_nodes$D == "Y" ] = "Duplication"
	all_nodes$Event[ all_nodes$D == "N" ] = "Speciation"
	

```


```{r show_result, echo=FALSE, cache=TRUE, message=F, warning=F}

	nodes_contrast = 
		all_nodes %>%
		filter( ! is.na(pic) ) %>%
		mutate( pic=abs(pic) ) %>%
		filter( ! is.na(D) )

	nodes_contrast %>%
		ggplot( aes( x=pic, y=..density.., col=Event ) ) + 
        geom_freqpoly( binwidth=.001, position="identity" ) +
        xlab("Phylogenetic independent \ncontrasts (PIC)") +
				xlim(0,0.075) + 
				theme_classic()
	
	wilcox_test_result = wilcox.test( nodes_contrast %>% filter(D=="Y") %>% .$pic, nodes_contrast %>% filter(D=="N") %>% .$pic, alternative="greater" )


```

> Figure XXKMRR | Density plot of the magnitude of phylogenetic independent contrast values following duplication and speciation events. These contrasts are not larger following duplication events, as predicted by the ortholog conjecture.


There were `r nodes_contrast %>% filter(D=="Y") %>% nrow()` duplication nodes and `r nodes_contrast %>% filter(D=="N") %>% nrow()` speciation nodes.

Given the ortholog conjecture, we would expect to see larger expression contrasts following duplication events than speciation events. Instead,we find no such distribution (Figure XXKMRR). The Wilcox rank test does not reject the null hypothesis that the rate of evolution following duplications is the same as or less than the rate following speciation (p value = `r wilcox_test_result$p.value`). Our results therefore find no evidence for the ortholog conjecture in this system.

We next examined the possibility that ascertainment biases were differentially impacting the inference of evolutionary change in differential expression following duplication and speciation events. We focused on two possible sources of bias - node depth and branch length. 

While the age of speciation nodes is constrained, duplication nodes can be much older and therefore have a wider range of ages (Supplementary Figure XXAscertainment a). Biological reasons for this include the fact that duplication events can be older than the root of the species tree, but speciation events cannot. There are also technical factors that can lead to an excess of duplication events deeper in the tree, including the impact of assembly errors that lead to the overestimation of deep duplications [@Hahn:2007cj]. If independent contrast values also tended to to be lower at greater node depth, it could interact with the preponderance of duplications at greater depth to create a pattern of lower contrasts following duplication events. When we remove duplication nodes that are older than the oldest speciation node, our results are unchanged and we cannot reject the null hypothesis that the rate of evolution following duplications is the same as or less than the rate following speciation (Supplementary Figure XXAscertainment b).

The independent contrast across a node is the amount of change observed between the daughter notes, scaled by the expected variance [Felsenstein:1985ua]. The expected variance is principaly determined by the lengths of the branches leading from the node to these daughters. The same observed difference between daughter nodes will lead to a larger contrast the shorter the branches are to these nodes, as this indicates a greater rate of evolutionary change. The expected variance of contrasts for speciation nodes is constrained by the branch lengths on the species tree, but the expected variance of contrasts for duplications can range much wider (Figure XXAscertainment c).

The fact that the expected variance of speciation contrasts is a subset of the range of the expected variance of duplication contrasts could lead to ascertainment biases. We therefore also examine only the contrasts that fell within in the range of expected variance seen for speciation contrasts, excluding duplication contrasts that fall outside of this range. This reanalysys does not reject the ortholog conjecture either (Figure XXAscertainment d).

The widespread invokation of the ortholog conjecture follows from the expectation that it is strong, widely distributed phenomenon. Our failure to detect a signature consistent with the ortholog conjecture suggests that it may not be, and that it should not be assumed in systems where it has not been explicitly tested. 


### Levin *et al.* reanalysis

![Figure Levin](levin_etal/Figure_levin.png?raw=true)

> Figure XXLevin | Distributions of pairwise similarity scores for each phase of development. Pairwise scores for the ctenophore are red. Wilcoxon test p-values for the significance of the differences between early-mid distributions and late-mid distributions are on the right. Model of variance, which is inversely related to similarity, is on the left. (a) The distributions as published. Low similarity (*i.e.*, high variance) in the mid phase of development was interpreted as support for an inverse hourglass model for the evolution of gene expression. The five least-similar mid phase scores were all from the ctenophore. Published KS p-values, based on duplicated data, are in parentheses. The inset ctenophore image is by S. Haddock from phylopic.org. (b) The distributions after the exclusion of the ctenophore. The early and mid phase distributions are not statistically distinct. This suggest a wine bottle model, with similar evolutionary variance at the early and mid phase and less at the late phase.

How changes in animal development relate to the evolution of animal diversity is a major question in evolutionary developmental biology (EvoDevo). To address this topic, Levin *et al.* [@Levin:2016bp] analyzed gene expression through the course of embryonic development for ten animal species, each from a different phylum. As their title ("The mid-developmental transition and the evolution of animal body plans") indicates, they arrived at two major conclusions. First, animal development is characterized by a well-defined mid-developmental transition. Second, this transition helps explain the evolution of features observed among distantly related animals. Specifically, they concluded that animals from different phyla exhibit an "inverse hourglass" model for the evolution of gene expression, where there is more evolutionary variance in gene expression at a mid phase of development than there is at early and late phases. Closely related animals have previously been described as having an hourglass model of gene expression, where evolutionary variance in expression is greater early and late in development than at the midpoint of development [@Kalinka:2010dw; @DomazetLoso:2010cn]. Levin *et al.* conclude that this contrast between distantly and closely related animals provides biological justification for the concept of phyla and may provide a long-sought operational definition of phyla. We previously described some concerns with their interpretations of these results [@Hejnol:2016hw]. 

Rather than support the inverse hourglass as a general pattern among distantly related species, our reanalyses of Levin *et al.* [@Levin:2016bp] indicate their existing data support the opposite conclusion - that the inverse hourglass is due to large differences specific to a single lineage, the ctenophore (comb jelly). Better understanding these differences will be of interest to future work, especially in light of current uncertainty regarding the phylogenetic placement of ctenophores and their unique biology [@Martindale:2015ve, @Dunn:2015ij]. 

To understand our concerns with their analyses, it is helpful to first outline Levin *et al.*'s published methods and results. Although there are mature methods and tools for statistical analysis of character evolution [@Garamszegi:2014ew] and gene expression [@Robinson:2009cw], Levin *et al.* [@Levin:2016bp] drew on neither of these very active areas of research and instead applied an *ad hoc* pairwise comparison of orthologous gene expression between species. They characterized each gene in each species as having expression that peaks in early, mid, or late temporal phase of development while the goodness of fit to these patterns was not considered, and alternative patterns were not evaluated. For each species pair, they then identified the orthologs shared by these species (shared orthologs vary from pair to pair). They then calculated a similarity score for each temporal phase for each species pair based on the fraction of genes that exhibited the same patterns in each species. The distributions of similarity scores are plotted in their [Figure 4d](http://www.nature.com/nature/journal/v531/n7596/fig_tab/nature16994_F4.html), and their Kolmogorov–Smirnov (KS) tests indicated that the early distribution and late distribution were each significantly different from mid distribution (P < 10<sup>-6</sup> and P < 10<sup>-12</sup>, respectively). This is the support they presented for the inverse hourglass model.

We began by examining the matrix of pairwise comparisons that their KS tests and Figure 4d are based on. As this figure shows (regenerated here as our Figure XXLevin a), there is not a strong distinction between these temporal phases - the distributions for the early, mid, and late distributions overlap considerably. The medians are close in value, and the range of mid phase includes the entire range of the early phase. Further inspection (Figure XXLevin a) reveals that the mid phase distribution has several outliers with very low similarity scores. We found that all five of the lowest values in the mid phase distribution are for pairwise comparisons that include the ctenophore (Figure XXLevin a). When the nine pairwise comparisons that include the ctenophore are removed, the differences between the early and mid phase distributions are greatly reduced (Figure XXLevin b).

We also found several problems with the statistical tests that were used to evaluate the inverse hourglass hypothesis. First, in the published analyses every data point was included twice because both reciprocal comparisons (which have the same values) were retained. For example, there is both a nematode to arthropod comparison and an arthropod to nematode comparison. As a consequence, there are 90 entries for the 45 pairwise comparisons, and by doubling the data the significance of the result appears much stronger than it actually is. After removing the duplicate values, the p values are far less significant, 0.002 for the early-mid comparison and on the order of 10<sup>-6</sup> for early-late. Second, the test they used (KS test) is not appropriate for the hypothesis they seek to evaluate. The KS test does not just evaluate whether one distribution is greater than the other, it also tests whether the shape of the distributions are the same. In addition, the samples in this dataset are matched (*i.e.*, for each pairwise comparison there is a early, mid, and late expression value), which the KS test does not take into account. The Wilcoxon test is instead appropriate in this case. After removing the duplicate scores, removing the ctenophore, and applying the Wilcoxon test, there is no significant difference between the early phase and mid phase distributions (P  = 0.1428 for the early-mid comparison and P < 10<sup>-5</sup> for the late-mid comparison), and the inverse hourglass turns into a wine bottle (Figure XXLevin b). 

Our results highlight the importance of explicitly accounting for phylogenetic relationships when studying character evolution, including developmental [@Telford:2003tsa] and functional genomic traits [@Hejnol:2015gr]. This is particularly true for evolutionary analyses of quantitative gene expression [@Dunn:2013cpa]. Pairwise comparisons result in the same evolutionary changes being counted multiple times in each species pair, as for the ctenophore comparisons that were mistaken for a global pattern here. This is a well understood property of pairwise comparisons that has been specifically addressed by phylogenetic comparative methods [@Felsenstein:1985ua]. This particular case illustrates the shortcomings of *ad hoc* rank-based methods [@DomazetLoso:2010cn] and pairwise comparisons [@Levin:2016bp] for the study of the evolution of gene expression. Phylogenetic analyses of character evolution provide not only a way to avoid these problems, but a richer context for interpreting the biological implications of the results.

While demonstrate a common problem with pairwise comparisons in Levin *et al.* dataset, we did not perform a phylogenetic reanalysis of this study, as we did for the KMRR study. This is because the similarity metric computed in the pairwise comparisons of Levin *et al.* are based on different genes for different species pairs. This means that there is not a trait that can be mapped onto the tree, as KMRR's tau can be. A full phylogenetic reanalysis would of course be possible be using upstream analysis products to rederive a new expression summary statistic that was suitable for mapping onto a phylogenetic tree.


## Discussion

### Comparative methods in functional genomics

Some of the most widely used phylogenetic comparative methods [1, 5] are directly applicable to CFG data. There are also unique challenges that will need to be addressed at this new intersection, though. One of the greatest challenges is that most phylogenetic comparative methods have been developed to address problems with many more species (e.g., dozens or more) relative to the number of traits being examined (e.g., 2-5). In CFG analyses, there are often far fewer species (e.g., 3-10) because adding taxa is still expensive, but tens of thousands of traits. This creates challenges as the resulting covariance matrices are singular and, if not treated appropriately, imply many false correlations that are artefacts of project design. We outlined these challenges and potential solutions in the context of gene expression [31]. 

It is not enough to demonstrate the impact of phylogenetics on CFG analyses, these methods need to be made accessible to the FG community so they can be widely.

Most early phylogenetic comparative methods attempted to remove evolutionary signal to correct statistical tests for correlations between traits, while more recent methods tend to focus on testing hypotheses of evolutionary processes [34]. The application of this newer focus to pCFG provides an exciting opportunity to address long standing questions of broad interest, including the order of changes in FG traits and shifts in rates of evolution of one FG trait following changes in another trait. 

A small number of CFG studies have employed some phylogenetic comparative approaches [18–22]. For instance, a phylogenetic ANOVA [21] of the evolution of gene expression improves statistical power and drastically reduces the rate of false positives relative to pairwise approaches.

### Ortholog conjecture

The ortholog conjecture [@Nehrt:2011bm] is the proposition that orthologs (genes that diverged from each other due to a speciation event) have more similar attributes than do paralogs (genes that diverged from each other due to a gene duplication event). It has important biological and technical implications. It shapes our understanding of the functional diversity of gene families. It is used to relate findings from well-studied genes to related genes that haven't been investigated in detail. It can be applied to any trait of genes, from gene seqeunce to biochemical properties to expression. While the ortholog conjecture describes a specific pattern of functional diversity across genes, it can also be considered a hypothesis about the process of evolution. Specifically, it postulates that there is greater evolutionary change in gene attributes following a duplication event than a speciation event. 

Despite its importance, there have been relatively few tests of the ortolog conjecture. Previous work has shown that ontology annotations are not sufficient to test the ortholog conjecture [@Chen:2012jz; @Thomas:2012hz]. Analyses of domain structure were consistent with ortholog conjecture [@Forslund:2011cw]. There have been few tests of the ortholog conjecture with regards to gene expression [@Chen:2012jz].

Our reanalysis of KMRR using phylogenetic methods suggests that the ortholog conjectures is not a dominant pattern that is central to explaining the evolution of phenotypic diversity in gene families. This has impotant biological implications. It suggests that the mechanism of gene divergence (speciation versus duplication) may not have as strong an impact on phenotypic divergence as thought. It also has technical implications. Rather than focus on whether genes are orthologs or paralogs when attempting to predict function, it may be more effective to simply focus on how closely related or distantly related they are. Closely related paralogs, for example, may tend to have more similar phenotypes than more distantly related orthologs. 

## Conclusions

The fact that the first two CFG studies we reanalyzed show serious problems with pairwise comparisons indicates that there likely to be similar problems in other CFG studies, and that future CFG studies will be compromised if they continue to use pairwise methods. 

## Methods

All files needed to reexecute the anlyses presented in this document are available at https://github.com/caseywdunn/comparative_expression_2017. The most recent commit at the time of the analysis presented here was `r system("git log | head -n 1", intern=TRUE) %>% str_replace("commit ", "")`.


### KMRR reanalysis

The KMRR study [@KryuchkovaMostacci:2016iw] followed excellent practices in reproducibility. They posted all data and code needed to re-execute their analyses at figshare:  https://figshare.com/articles/Tissue-specificity_of_gene_expression_diverges_slowly_between_orthologs_and_rapidly_between_paralogs/3493010/2 . We slightly altered their `Rscript.R` to simplify file paths. This midified script and their data files are available in the gihub respository for this paper, as are the intermediate files that were generated by their analysis script that we used in our own analyses. We also obtained the `Compara.75.protein.nh.emf` gene trees [@Herrero:2016jj] from ftp://ftp.ensembl.org/pub/release-75/emf/ensembl-compara/homologies/ and place them in the same directory as this file.

Each of these gene trees includes includes homologous sequences derived from a common ancestral sequence. The tip nodes of a gene tree can include both orthogs and paralogs, and each internal node is an infered speciation or duplication event. Prior to phylogenetic analyses of gene expression, each tree was first pruned to include only tips for which expression data were available. The speciation nodes in the tree were then time calibrated so that branch lengths were consistent across trees. We then calculated the magnitude of the independent contrast for differential expression at each internal node in each of these trees. The independent contrast is the amount of change in expression observed along the branches that arise from the node, scaled by the expected variance [Felsenstein:1985ua]. The expected variance is determined mostly by the length of the branches, but can also be influenced by uncertainty regarding the reconstruction of differential expression at internal nodes. We took the absolute value of these contrasts, since the sign depends only on the arbitrary ordering of the nodes in the trees.

### Levin et al. reanalysis

Levin *et al.* helpfully provided data and clarification on methods. We obtained the matrix of pairwise scores that underlies their [Figure 4d](http://www.nature.com/nature/journal/v531/n7596/fig_tab/nature16994_F4.html) and confirmed we could reproduce their published results. We then removed duplicate rows, identified ctenophores as overrepresented among the low outliers in the mid-developmental transition column, and applied the Wilcoxon test in place of the Kolmogorov-Smirnov test. These data, our analysis code, and additional information are available in a git repository at [https://github.com/caseywdunn/levin2016](https://github.com/caseywdunn/levin2016). This repository includes our analysis notebook at [https://rawgit.com/caseywdunn/levin2016/master/reanalyses.html](https://rawgit.com/caseywdunn/levin2016/master/reanalyses.html).

## Ackowledgements

Thanks to XX. This work was supported by the National Science Foundation (DEB-1256695 and the Waterman Award).


## Supplementary Material

### KMRR Analyses


```{r filtered, cache=TRUE, echo=FALSE, fig.cap="XX_PIC_VAR", warning=FALSE, message=FALSE}

	# Examine by node age

  g_age =
  nodes_contrast %>%
    ggplot( aes( x=node_age, y=pic, col=Event ) ) + 
		  geom_point( alpha=0.2 ) + 
		  xlab( "Node age (millions of years)" ) + 
		  ylab("Phylogenetic independent contrasts (PIC)\n of differential expression") +
		  ylim(0,0.3) +
			xlim(0,1000) +
		  ggtitle( "(a)" ) +
		  theme_classic()	
		  #theme( legend.justification=c( 1, 1 ), legend.position=c( 1, 1 ) )


	max_speciation_age = 
    nodes_contrast %>% 
    filter( D=="N" ) %>%
    select( node_age ) %>%
    max( na.rm=TRUE )
	
	nodes_contrast_filtered_age = 
		nodes_contrast %>% 
		filter( node_age <= max_speciation_age )
	
	wilcox_filtered_age = 
		wilcox.test( 
			nodes_contrast_filtered_age %>% filter(D=="Y") %>% .$pic, 
			nodes_contrast_filtered_age %>% filter(D=="N") %>% .$pic, 
			alternative="greater" 
		)$p.value
	
	label_age = str_c( "Wilcoxon p=", wilcox_filtered_age )
	
	g_density_filtered_age =   
		nodes_contrast_filtered_age %>%
		ggplot( aes( x=pic, y=..density.., col=Event ) ) + 
      geom_freqpoly( binwidth=.001, position="identity" ) +
      xlab("Phylogenetic independent \ncontrasts (PIC)") +
			ylab("Density") +
			xlim(0,0.075) + 
			ggtitle( "(b)" ) +
			annotate("text", x = Inf, y = Inf, hjust="right", vjust="top", label=label_age ) +
			theme_classic()


	# Examine by expected variance (branch length)
  g_var_exp =
    nodes_contrast %>%
      ggplot( aes( x=var_exp, y=pic, col=Event ) ) + 
      geom_point( alpha=0.2 ) + 
      xlab( "Expected variance, \n depends mostly on branch length" ) + 
      ylab("Phylogenetic independent contrasts (PIC)\n of differential expression") +
      ylim(0,0.3) +
			xlim(0,2000) +
      ggtitle( "(c)" ) +
      theme_classic()	
      # theme( legend.justification=c( 1, 1 ), legend.position=c( 1, 1 ) )

  var_exp_max = 
    nodes_contrast %>% 
    filter( D=="N" ) %>%
    select( var_exp ) %>%
    max( na.rm=TRUE )

  var_exp_min = 
    nodes_contrast %>% 
    filter( D=="N" ) %>%
    select( var_exp ) %>%
    min( na.rm=TRUE )
  
  nodes_contrast_filtered_var = 
    nodes_contrast %>% 
      filter ( var_exp >=var_exp_min & var_exp <=var_exp_max )
  
  
	wilcox_filtered_var = 
	wilcox.test( 
		nodes_contrast_filtered_var %>% filter(D=="Y") %>% .$pic, 
		nodes_contrast_filtered_var %>% filter(D=="N") %>% .$pic, 
		alternative="greater" 
	)$p.value
	
	label_var = str_c( "Wilcoxon p=", wilcox_filtered_var )
	
  g_density_filtered_var =   
    nodes_contrast_filtered_var  %>% 
    ggplot( aes( x=pic, y=..density.., col=Event ) ) + 
      geom_freqpoly( binwidth=.001, position="identity" ) + 
      ggtitle( "(d)" ) +
      theme( legend.justification=c( 1, 1 ), legend.position=c( 1, 1 ) ) +
      xlim(0,0.075) + 
			theme_classic() +
  		annotate("text", x = Inf, y = Inf, hjust="right", vjust="top", label=label_var ) +
      xlab("Phylogenetic independent contrasts (PIC)\n of differential expression") +
  		ylab("Density")


  grid.arrange(
  	g_age,
  	g_density_filtered_age,
		g_var_exp,
		g_density_filtered_var,
		ncol = 2
	)
```


> Supplementary Figure XXAscertainment | Investigation of possible assertainment biases. (a) Magnitude of independent contrasts ploted against node age. Speciation nodes are calibrated to particular times, whereas duplication nodes have a wider range. (b) Density plot of contrasts for only the nodes that have an age less than or equal to the maximum age of speciation nodes. (c) Magnitude of independent contrasts ploted against expected variance, which is largely determined by branch lengths. Contrasts for speciation nodes have a narrower range of expected variance than do contrasts for duplication nodes. (d) Density plot of contrasts for only the nodes that have expected variance within the range of contrasts across speciation nodes.

```{r pairwise, echo=FALSE, cache=TRUE, message=F, warning=F}
	get_pairwise_summary = function ( nhx ) {
		
		# Create a tibble with one row for each pairwise combination of tip nodes
		ntips = length( nhx@phylo$tip.label )
		D = 1:ntips %>% combn( 2 ) %>% t() %>% as_tibble()
		names( D ) = c( "tip_a", "tip_b" )
		
		# Add a colmn with the gene tree hash 
		D %<>% mutate( gene_tree = digest( nhx ) )
		
		# Calculate the phenotypic difference between the tips
		D$difference = 
			apply(
				D,
				1,
				function(y) {
					tau_values = nhx@data$Tau[ c( as.integer(y['tip_a']), as.integer( y['tip_b'] ) ) ]
					max( tau_values ) - min( tau_values )
				}
			)
		
		# For each row, bet the most recent common ancestor of the pairwise tip combination
		D$mrca = 
			apply(
				D, 
				1, 
				function(y) ape::getMRCA( 
					nhx@phylo, 
					c( as.integer(y['tip_a']), as.integer(y['tip_b'] ))
				)
			) %>%
			as.integer()
		
		# Merge data on this mrca to the node
		D %<>% left_join(
			nhx@data %>%
				select( D, node, node_age, pic ),
			by = c( "mrca" = "node")
				
		)
		
		# Calculate the distance between the tips, which for an ultrametric tree 
		# is twice the age of their most recent common ancestor
		D %<>% mutate( distance = 2 * node_age )
		
		return( D )
		
	}

	# Build a tibble of all pairwise comparisons between tips
	pairwise_summary = 
		mclapply( gene_trees_pic, get_pairwise_summary, mc.cores=cores ) %>% 
		bind_rows()
	
	
	pairwise_summary %>% 
		filter( ! is.na(D) ) %>%
  	ggplot(aes(x=distance, y=difference, color=D)) + 
  		geom_point(alpha=0.1) +
  		geom_smooth(method=lm, se=FALSE) +
			xlim( 0, 2000 )
	
```

> Supplementary Figure XXPairwise | Pairwise comparisons

## References
